selftests/unit/test_utils_iso9660.py

```diff
@@ -0,0 +1,119 @@
+"""
+Verifies the avocado.utils.iso9660 functionality
+"""
+import os
+import shutil
+import sys
+import tempfile
+
+from avocado.utils import iso9660, process
+
+
+if sys.version_info[:2] == (2, 6):
+    import unittest2 as unittest
+else:
+    import unittest
+
+
+class BaseIso9660(unittest.TestCase):
+
+    """
+    Base class defining setup and tests for shared Iso9660 functionality
+    """
+
+    def setUp(self):
+        self.iso_path = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                     os.path.pardir, ".data",
+                                                     "sample.iso"))
+        self.iso = None
+        self.tmpdir = tempfile.mkdtemp(prefix="avocado_" + __name__)
+
+    def basic_workflow(self):
+        """
+        Check the basic Iso9660 workflow
+        """
+        self.assertEqual(self.iso.read("file"),
+                         "file content\n")
+        dst = os.path.join(self.tmpdir, "file")
+        self.iso.copy(os.path.join("Dir", "in_dir_file"), dst)
+        self.assertEqual(open(dst).read(), "content of in-dir-file\n")
+        self.iso.close()
+        self.iso.close()    # check that double-close won't fail
+
+    @unittest.skipIf(os.getuid(), "This test is only available to root")
+    def mnt_dir_workflow(self):
+        """
+        Check the mnt_dir functionality
+        """
+        base = self.iso.mnt_dir
+        os.path.isdir(os.path.join(base, "Dir"))
+        self.assertEqual(open(os.path.join(base, "file")).read(),
+                         "file content\n")
+        self.assertEqual(open(os.path.join(base, "Dir", "in_dir_file")).read(),
+                         "content of in-dir-file\n")
+        self.iso.close()
+        self.assertFalse(os.path.exists(base), "the mnt_dir is suppose to be "
+                         "destroyed after iso.close()")
+
+    def tearDown(self):
+        if self.iso is not None:
+            self.iso.close()
+        shutil.rmtree(self.tmpdir)
+
+
+class BasicWorkflows(object):
```

Comment:
The test_mnt_dir is separate only because it might not be available somewhere. I could have fit it into one test, but then it'd be not obvious, whether this optional part was skipped or not.